doc/howtowritearecogmethod.autodoc

```diff
@@ -0,0 +1,181 @@
+@Chapter How to write a recognition method
+
+This chapter explains how to integrate a newly developed group recognition
+method into the framework provided by the recog package.
+
+TODO:
+Refer to Chapter <Ref Chap="methsel"/> for an explanation of methods.
+There are leaf methods and split methods.
+The next two sections describe how to implement leaf and split methods
+respectively, and include example code.
+
+@Section Leaf methods
+
+A leaf method must at the very least do the following:
+
+- Provide the order of the recognized group via `SetSize(ri, NNN)`.
+- Provide a set of SLPs which map the original generators $X$ to the nice generators $Y$, as entry for the attribute `slptonice`.
```

Comment:
Maybe for consistency, say that this can be set via `Setslptonice(ri, slp)`

Reply:
If the reader does not know how to set an attribute in GAP, I think they need to read some other tutorial first :-).

However, we could point out that all of these are show in the example later on, and maybe also add a Ref to that example